app/containers/LoginCallback/sagas.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * 1. sets up auth library and waits for auth token
+ * 2. gets user profile
+ */
+
+import { call, put } from 'redux-saga/effects';
+import { browserHistory } from 'react-router';
+import { loginCallback, getProfile } from './lib';
+import { loginSuccess, loginError } from './actions';
+
+function* loginCallbackSaga() {
+  // TODO take the proper event so that this saga is run exactly once (when loading route)
```

Comment:
OK somehow it's because I `browserHistory.push` within that saga, need to extract